Add keys for Latvian

eandersons opened the pull request:
This is my shot at adding Latvian specific characters (using cedille, caron (new) and macron (new) accents). Are these changes enough? I have no clue on how to build .apk for testing.

Also, I am not sure about lines like the following in `res/xml/qwerty.xml`:

```xml
    <key key0="s" key1="accent_caron" key1="accent_ring" key3="ß" />
```
Will this work? Or is there another way to define multiple accents for one key? Or should I add additional accent as a new key (for example, as `key2="accent_caron"` in this example)?

Julow commented on srcs/juloo.keyboard2/KeyValue.java:
```diff
@@ -40,6 +40,8 @@
   public static final int FLAG_ACCENT4 = (1 << 19); // Tilde
   public static final int FLAG_ACCENT5 = (1 << 20); // Cédille
   public static final int FLAG_ACCENT6 = (1 << 21); // Tréma
+  public static final int FLAG_ACCENT6 = (1 << 22); // Caron
+  public static final int FLAG_ACCENT6 = (1 << 23); // Macron
```
These are the next free flags:
```suggestion
  public static final int FLAG_ACCENT_CARON = (1 << 26); // Caron
  public static final int FLAG_ACCENT_MACRON = (1 << 27); // Macron
```
Also, let's start writing the accent names.

Julow reviewed (commented):
Hi! Thanks !
There's a few things missing in your patch right now: the effect of accent flags on keys must be defined in `KeyModifier.java` and the accent keys must be added on every keyboard layouts, only at one place in each layout.
See this commit for an example https://github.com/Julow/Unexpected-Keyboard/commit/573c13fb825f41a926dca838ef5a396ccce6a3eb

Julow commented:
I just wrote a guide on how to build the app: [CONTRIBUTING.md](https://github.com/Julow/Unexpected-Keyboard/blob/master/CONTRIBUTING.md)

eandersons commented:
I added missing changes and fixed previous work.
Only thing I am not sure of is the order of keys where new accents were added. I sorted them from my point of view, but I guess there would be people that might prefer a different order.

I could not build .apk to test my changes because of the following error:

```
$ make
mkdir -p _build/class
javac -d _build/class -source 1.7 -target 1.7 -encoding utf8 \
        -classpath /home/edgars/Android/Sdk/platforms/android-30/android.jar: \
        -sourcepath srcs:_build/gen \
        srcs/juloo.keyboard2/EmojiGroupButtonsBar.java srcs/juloo.keyboard2/Theme.java srcs/juloo.keyboard2/EmojiGridView.java srcs/juloo.keyboard2/KeyValue.java srcs/juloo.keyboard2/EmojiTypeButton.java srcs/juloo.keyboard2/Emoji.java srcs/juloo.keyboard2/SettingsActivity.java srcs/juloo.keyboard2/KeyboardData.java srcs/juloo.keyboard2/EmojiKeyButton.java srcs/juloo.keyboard2/KeyEventHandler.java srcs/juloo.keyboard2/Keyboard2View.java srcs/juloo.keyboard2/Config.java srcs/juloo.keyboard2/KeyModifier.java srcs/juloo.keyboard2/Keyboard2.java srcs/juloo.common/IntSlideBarPreference.java srcs/juloo.common/SlideBarPreference.java _build/gen/juloo/keyboard2/R.java
make: javac: No such file or directory
make: *** [Makefile:105: _build/classes.dex] Error 127
```

Julow commented:
You need Java and the Android SDK. If you are on Ubuntu, `sudo apt install openjdk-8-jdk android-sdk`, then you'll need to use the `androidsdk` graphical program to install the right "platform" and "build tools".
I strongly suggest that you install [Nix](https://nixos.org/download.html) and run `nix-shell ./shell.nix` to be sure to have a working environment. Both ways will download hundreds of megabytes.

eandersons commented:
I built the app for testing using Nix, and I finally understood the concept, although it is not what I had in my mind when I started working on changes in this pull request, but I hope that somebody will find this useful.

I have made the final adjustments. @Julow, will you please review changes?

Julow reviewed (commented):
Awesome :)

The wikipedia page for [Latvian](https://en.wikipedia.org/wiki/Latvian_language) talks about `ā`, `ī` and `ū`, are they rarely used ?

eandersons commented:
No, `ā`, `ī` and `ū` are used quite often.

Julow commented:
I forgot, one of the initial design decision I took while making this app is that all the layouts should have all the keys. This means the new accents should be on the other layouts too (azerty, qwertz).

The main reason for this is because this app is intended for programmers, which tend to use qwerty. This rule doesn't make much sense in the other direction, however. Why allocate space for the Latvian and Swedish accents on the French or German layout ?
I'm still attached to having every languages work with the qwerty layout.

Let's merge this and continue thinking about this in https://github.com/Julow/Unexpected-Keyboard/pull/29.

> The wikipedia page for Latvian talks about ā, ī and ū, are they rarely used ?

> No, ā, ī and ū are used quite often.

Oups, it's one of the accents you added. Ignore this.
